projects/plugins/wpcomsh/private-site/private-site.php

```diff
@@ -770,6 +770,19 @@ function private_robots_txt() {
 	return "User-agent: *\nDisallow: /\n";
 }
 
+/**
+ * Renames the "Subscriber" role to "Viewer".
+ *
+ * @param WP_Roles $roles WP_Roles object.
+ */
+function rename_subscriber_role_to_viewer( $roles ) {
+	if ( site_is_private() && isset( $roles->roles['subscriber'] ) ) {
+		$roles->roles['subscriber']['name'] = _x( 'Viewer', 'User role', 'wpcomsh' );
+		$roles->role_names['subscriber']    = _x( 'Viewer', 'User role', 'wpcomsh' );
+	}
+}
```

Comment:
```suggestion
function rename_subscriber_role_to_viewer( $roles ) {
	static $being_called;
	if ( 1 === $being_called ) {
		// Using locale related functions inits the roles system, which calls this function.
		// This breaks the loop.
		return;
	}
	$being_called = 1;
	if ( site_is_private() && isset( $roles->roles['subscriber'] ) ) {
		
		$roles->roles['subscriber']['name'] = _x( 'Viewer', 'User role', 'wpcomsh' );
		$roles->role_names['subscriber']    = _x( 'Viewer', 'User role', 'wpcomsh' );		
	}
	$being_called = 0;
}
```

There's probably a better way to do this (immediately deregister this hook?), but this stops the OOM and gives me 'Viewer'.